### PR TITLE
feat: add Git::Repository::Configuring#global_config facade and deprecated aliases

### PR DESCRIPTION
**Pull request by jcouball**
Review our [guidelines for contributing](https://github.com/ruby-git/ruby-git/blob/main/CONTRIBUTING.md) to this repository.

# Description

Implements the `Git::Repository::Configuring#global_config` facade and deprecated
backward-compatibility aliases as specified in `redesign/c1c2_bucket6_lib_orphans.md §3.3`
and the binding constraint in `§4.1` (no `:system`, no `:file` on `global_config`).

## What was changed

**`lib/git/repository/configuring.rb`**

- Added `global_config(name = nil, value = nil)` public facade method with three-way
  dispatch targeting the user's global config (`~/.gitconfig`):
  - No args → returns `Hash{String => String}` of all global config entries
  - One arg → returns `String` value for the named key
  - Two args → writes the value and returns `Git::CommandLineResult`
- Added three private helpers in the `Private` module:
  `global_config_get`, `global_config_list`, `global_config_set` — each passes
  `global: true` to the existing `ConfigOptionSyntax` command classes
- Added three deprecated forwarding methods `global_config_get(name)`,
  `global_config_list`, and `global_config_set(name, value)` that emit
  `Git::Deprecation.warn` and delegate to `global_config`
- Updated the module-level YARD summary to mention `global_config`
- Full three-`@overload` YARD docs on `global_config`

**`lib/git/base.rb`**

- Added `Git::Base` delegators for all four methods: `global_config`,
  `global_config_get`, `global_config_list`, `global_config_set`

**`redesign/c1c2_audit.md`**

- §7.3: Updated the three `global_config_*` rows from `🔍 human decision` to
  `✅ promoted — global_config facade + deprecated aliases in
  Git::Repository::Configuring + Git::Base delegators added (PR 5h-3)`
- §7.5: Decremented the `🔍 human decision` count from 16 to 13

**`redesign/c1c2_bucket6_lib_orphans.md`**

- §3.3: Marked as implemented in PR 5h-3

## Design decisions

- **Narrow signature** (`name = nil, value = nil` only): §4.1 is binding —
  no `:system` option, no `:file` option. Global mode and `--file` are mutually
  exclusive in git itself.
- **Deprecated aliases** delegate to `global_config`, not to `Private` helpers
  directly, so the delegation chain is clear and easy to trace.
- **UPGRADING.md** already contained the needed rows from a prior session; no
  changes were required.

## Test coverage

- **Unit tests** (`spec/unit/git/repository/configuring_spec.rb`):
  - 12 examples for `#global_config` (list/get/set paths, edge cases: `=` in values,
    valueless keys, empty stdout, key-not-found raises `Git::FailedError`)
  - 2 examples each for `#global_config_get`, `#global_config_list`,
    `#global_config_set` (deprecation warning emitted; delegation returns correct value)
- **Unit tests** (`spec/unit/git/base_spec.rb`):
  - 1 example each for the four `Git::Base` delegators (4 total)
- **Integration tests** (`spec/integration/git/repository/configuring_spec.rb`):
  - `global_config` (no args) returns a `Hash` with string keys and values
  - `global_config('user.name')` returns a non-empty `String`

## Checklist

- [x] I reviewed and applied the project's [AI Policy](../AI_POLICY.md). I understand
  and verify any AI-assisted changes included in this PR and ensured they meet
  quality, security, and licensing standards.
- [x] I ran `bundle exec rake` locally on this branch and it passed (see
  [Local development setup](../CONTRIBUTING.md#local-development-setup)).
- [x] Tests added/updated as needed.
- [x] Documentation updated where applicable (README and/or YARD).

### CODE REVIEW

File: lib/git/repository/configuring.rb [hunk 1]
```diff
@@ -81,6 +83,88 @@ def config(name = nil, value = nil, options = {})
...
+      #
+      def global_config(name = nil, value = nil)
+        if name && value
+          Private.global_config_set(@execution_context, name, value)
+        elsif name
+          Private.global_config_get(@execution_context, name)
+        else
+          Private.global_config_list(@execution_context)
+        end
+      end
```
[the remark this answers is not in the document]
Comment (jcouball) [reply lib/git/repository/configuring.rb:141]: Fixed: changed `if name && value` to `if !name.nil? && !value.nil?`. Added a unit test for the `value is false` case confirming it routes to Set, not Get.

File: spec/integration/git/repository/configuring_spec.rb [hunk 2]
```diff
@@ -45,4 +45,27 @@
...
+        expect(result.values).to all(be_a(String))
+      end
+    end
+
+    context 'when called with a name' do
+      it 'returns a non-empty String for user.name' do
+        result = described_instance.global_config('user.name')
+        expect(result).to be_a(String)
+        expect(result).not_to be_empty
+      end
```
[the remark this answers is not in the document]
Comment (jcouball) [reply spec/integration/git/repository/configuring_spec.rb]: Fixed: the test now creates an isolated `Dir.mktmpdir` home with a pre-written `.gitconfig` and overrides `ENV['HOME']` (using `Dir.home` to save the original per RuboCop Style/EnvHome). Tests assert against the known value `'GlobalUser'` rather than relying on the real `~/.gitconfig`.

File: spec/integration/git/repository/configuring_spec.rb [hunk 3]
```diff
@@ -45,4 +45,37 @@
...
+        "[user]\n\tname = GlobalUser\n\temail = global@example.com\n"
+      )
+      @original_home = Dir.home
+      ENV['HOME'] = tmp_home
+    end
+
+    after do
+      ENV['HOME'] = @original_home
+      FileUtils.rm_rf(tmp_home)
+    end
```
[the remark this answers is not in the document]
Comment (jcouball) [reply spec/integration/git/repository/configuring_spec.rb]: Fixed: switched to `GIT_CONFIG_GLOBAL` isolation matching the project pattern in `spec/integration/git/lib/config_spec.rb`. The `around` block sets `GIT_CONFIG_GLOBAL` to a temp file in `repo_dir`, restores it in `ensure`, and each context sets the expected values via `global_config(name, value)` before reading them — no `HOME` mutation needed.

[the remark this answers is not in the document]
Comment (jcouball) [reply spec/integration/git/repository/configuring_spec.rb]: Fixed: updated both the public `@overload global_config(name, value)` doc and the private `global_config_set` helper doc from `[String]` to `[#to_s]` with a note that the value is converted via `#to_s` before being passed to git.

File: lib/git/repository/configuring.rb [hunk 4]
```diff
@@ -81,6 +83,88 @@ def config(name = nil, value = nil, options = {})
...
+      #   @raise [Git::FailedError] if git exits with a non-zero exit status
+      #
+      # @overload global_config(name, value)
+      #
+      #   @example Set the global committer name
+      #     repo.global_config('user.name', 'Alice')
+      #
+      #   @param name [String] the dotted config key to write (e.g. `"user.name"`)
+      #
+      #   @param value [String] the value to assign
```
[the remark this answers is not in the document]
Comment (jcouball) [reply lib/git/repository/configuring.rb]: Fixed: updated both the public `@overload global_config(name, value)` doc and the private `global_config_set` helper doc from `[String]` to `[#to_s]` with a note that the value is converted via `#to_s` before being passed to git.

File: spec/integration/git/repository/configuring_spec.rb (hunk 3 above)
[the remark this answers is not in the document]
Comment (jcouball) [reply spec/integration/git/repository/configuring_spec.rb]: Fixed: changed `~/.gitconfig` to `git config --global` in the module-level doc (line 13) to accurately describe the global config scope rather than a specific file path.

File: lib/git/repository/configuring.rb [hunk 5]
```diff
@@ -7,8 +7,10 @@ module Git
   class Repository
     # Facade methods for reading and writing git configuration
     #
-    # Provides the {#config} method, which dispatches to read a single entry,
-    # list all entries, or write a value depending on the arguments supplied.
+    # Provides the {#config} and {#global_config} methods, which dispatch to read a
+    # single entry, list all entries, or write a value depending on the arguments
+    # supplied. {#config} targets the repository's local config; {#global_config}
+    # targets the user's global config (`~/.gitconfig`).
```
[the remark this answers is not in the document]
Comment (jcouball) [reply lib/git/repository/configuring.rb]: Fixed: changed `~/.gitconfig` to "git's global config scope (`git config --global`)" so the doc accurately reflects that the scope may resolve to XDG paths or be overridden by `GIT_CONFIG_GLOBAL`.

File: lib/git/repository/configuring.rb [hunk 6]
```diff
@@ -81,6 +83,89 @@ def config(name = nil, value = nil, options = {})
         end
       end
 
+      # Read or write a global git configuration entry
+      #
+      # Dispatches to one of three modes depending on the arguments supplied,
+      # targeting the user's global git config (`~/.gitconfig`):
```
[the remark this answers is not in the document]
Comment (jcouball) [reply lib/git/repository/configuring.rb]: Fixed: changed `~/.gitconfig` to "git global config scope (`git config --global`)" in the `#global_config` method doc to accurately reflect that the scope may resolve to XDG paths or be overridden by `GIT_CONFIG_GLOBAL`.

File: lib/git/repository/configuring.rb [hunk 7]
```diff
@@ -81,6 +83,89 @@ def config(name = nil, value = nil, options = {})
...
+          Private.global_config_list(@execution_context)
+        end
+      end
+
+      # @deprecated Use {#global_config} instead.
+      def global_config_get(name)
+        Git::Deprecation.warn(
+          'global_config_get is deprecated and will be removed in a future version. ' \
+          'Use global_config(name) instead.'
+        )
```
[the remark this answers is not in the document]
Comment (jcouball) [reply lib/git/repository/configuring.rb:148]: Fixed: updated the deprecation message to `'Git::Repository#global_config_get is deprecated...'`. Updated the unit test to assert the fully-qualified form is included in the warning.

File: lib/git/repository/configuring.rb [hunk 8]
```diff
@@ -81,6 +83,89 @@ def config(name = nil, value = nil, options = {})
...
+        global_config(name)
+      end
+
+      # @deprecated Use {#global_config} instead.
+      def global_config_list
+        Git::Deprecation.warn(
+          'global_config_list is deprecated and will be removed in a future version. ' \
+          'Use global_config instead.'
+        )
```
[the remark this answers is not in the document]
Comment (jcouball) [reply lib/git/repository/configuring.rb:157]: Fixed: updated the deprecation message to `'Git::Repository#global_config_list is deprecated...'`. Updated the unit test to assert the fully-qualified form.

File: lib/git/repository/configuring.rb [hunk 9]
```diff
@@ -81,6 +83,89 @@ def config(name = nil, value = nil, options = {})
...
+        global_config
+      end
+
+      # @deprecated Use {#global_config} instead.
+      def global_config_set(name, value)
+        Git::Deprecation.warn(
+          'global_config_set is deprecated and will be removed in a future version. ' \
+          'Use global_config(name, value) instead.'
+        )
```
[the remark this answers is not in the document]
Comment (jcouball) [reply lib/git/repository/configuring.rb:166]: Fixed: updated the deprecation message to `'Git::Repository#global_config_set is deprecated...'`. Updated the unit test to assert the fully-qualified form.

File: redesign/c1c2_audit.md [hunk 10]
```diff
@@ -441,7 +441,7 @@ upgrade notes as "unsupported; remove any `g.lib.X` calls."
 | ✅ promote (repo already had it, `Git::Base` delegator added — PR 2d; or alias added) | 26 |
 | ⬜ promote (new facade work required) | 1 |
 | ❌ remove (internal plumbing) | 12 |
-| 🔍 human decision | 14 |
+| 🔍 human decision | 11 |
```
[the remark this answers is not in the document]
Comment (jcouball) [reply redesign/c1c2_audit.md]: Fixed: increased the `✅ promote` count from 26 to 29 (+3 for `global_config_get/list/set` promoted in this PR). The 🔍 count went 14→11 on this PR (14 reflects main's state after a prior PR; the original audit had 16, reduced by 2 in a prior merge before this one rebased). Both counts are now consistent: 3 items moved from 🔍 to ✅.

File: lib/git/repository/configuring.rb [hunk 11]
```diff
@@ -7,8 +7,10 @@ module Git
   class Repository
     # Facade methods for reading and writing git configuration
     #
-    # Provides the {#config} method, which dispatches to read a single entry,
-    # list all entries, or write a value depending on the arguments supplied.
+    # Provides the {#config} and {#global_config} methods, which dispatch to read a
+    # single entry, list all entries, or write a value depending on the arguments
+    # supplied. {#config} targets the repository's local config; {#global_config}
+    # targets the user's global config scope (`git config --global`).
```
[the remark this answers is not in the document]
Comment (jcouball) [reply lib/git/repository/configuring.rb]: Fixed: replaced "targets the repository's local config" with "uses git's default config scope (reads from the full resolution chain; set operations write to the repository's `.git/config`)" — which accurately reflects that `config_get`/`config_list` use no `--local` flag and read from system/global/local/worktree as git resolves them.

File: redesign/c1c2_audit.md [hunk 12]
```diff
@@ -438,10 +438,10 @@ upgrade notes as "unsupported; remove any `g.lib.X` calls."
 
 | Status | Count |
 |--------|-------|
-| ✅ promote (repo already had it, `Git::Base` delegator added — PR 2d; or alias added) | 26 |
+| ✅ promote (repo already had it, `Git::Base` delegator added — PR 2d; or alias added) | 29 |
 | ⬜ promote (new facade work required) | 1 |
 | ❌ remove (internal plumbing) | 12 |
-| 🔍 human decision | 14 |
+| 🔍 human decision | 11 |
```
[the remark this answers is not in the document]
Comment (jcouball) [reply redesign/c1c2_audit.md:444]: Fixed: updated the recommendation text from "4 new facade promotions and 16 human-decision items" to "1 new facade promotion and 11 human-decision items" to match the current table counts.
